sdk/storage/azure-storage-common/src/storage_per_retry_policy.cpp

```diff
@@ -37,11 +37,11 @@ namespace Azure { namespace Storage { namespace _internal {
     }
     else
     {
-      auto currentTimepoint = std::chrono::system_clock::now();
-      int64_t numSeconds
-          = std::chrono::duration_cast<std::chrono::seconds>(
-                std::chrono::system_clock::time_point(cancelTimepoint) - currentTimepoint)
-                .count();
+      auto currentTimepoint = DateTime(std::chrono::system_clock::now());
+      int64_t numSeconds = (cancelTimepoint > currentTimepoint)
+          ? std::chrono::duration_cast<std::chrono::seconds>(cancelTimepoint - currentTimepoint)
+                .count()
+          : -1;
```

Comment:
Why -1 as the alternative sentinel?

Reply:
It's not a sentinel; the here is to protect against underflow.  In this case, any non-positive value is as good as another.